lib/branca.rb

```diff
@@ -23,14 +24,18 @@ def encode(message, timestamp = Time.now.utc)
       BaseX::Base62.encode(raw_token)
     end
 
-    def decode(token)
+    def decode(token, ttl: self.ttl, secret_key: self.secret_key)
       header, bytes = token_explode(token)
       version, timestamp, nonce = header_explode(header)
 
       raise VersionError unless version == VERSION
-      raise ExpiredTokenError if (timestamp + Branca.ttl) < Time.now.utc.to_i
+      raise ExpiredTokenError if (timestamp + ttl) < Time.now.utc.to_i
 
+      cipher = RbNaCl::AEAD::XChaCha20Poly1305IETF.new(secret_key)
       message = cipher.decrypt(nonce, bytes.pack('C*'), header.pack('C*'))
+    rescue RbNaCl::CryptoError
```

Comment:
Great ideia!

Reply:
I do think this is _technically_ not backwards compatible though. Your call if you want to keep and release with a major version perhaps

Reply:
Because if there is code which does:

```
begin
  Branca.decode('ciphertext')
rescue RbNaCl::CryptoError
  # ...
end
```

...then their exception handler will be skipped and the stack will jump much higher.

Reply:
I understand now, however DecodeError will handle that! I will generate a new version and publish to rubygems.